kiali/kiali
# Added GatewayClassName validations in K8sGateways

Issue https://github.com/kiali/kiali/issues/6429

Added a new validation check for K8sGateway object's gatewayClassName field.
When a referenced GatewayClass does not exist in cluster, an Error message is shown "K8s GatewayClass not found"

![Screenshot from 2023-09-12 17-09-21](https://github.com/kiali/kiali/assets/604313/f2f16ea0-000b-4f3c-a893-15754eb2f8fd)

How to test:
Configure Kiali to have several Gateway implementations:
```
    gateway_api_classes:
      - name: Contour
        class_name: contour
      - name: Istio
        class_name: istio
      - name: Cilium
        class_name: cilium
```
Configure K8sGateway in cluster: https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/
Do not create a GatewayClass "cilium".
Create K8sGateway which refers to 'cilium' gatewayClassName. Verify that it shows a validation error on that object list and details pages.
Create K8sGateway which refers to 'istio' gatewayClassName. Verify that no error is shown for this object.
